kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyReconciler.java

```diff
@@ -85,26 +122,51 @@ public ErrorStatusUpdateControl<KafkaProxy> updateErrorStatus(KafkaProxy primary
     }
 
     private static KafkaProxy buildStatus(KafkaProxy primary,
+                                          Context<KafkaProxy> context,
                                           @Nullable Exception exception) {
+        if (exception instanceof AggregatedOperatorException aoe && aoe.getAggregatedExceptions().size() == 1) {
+            exception = aoe.getAggregatedExceptions().values().iterator().next();
+        }
+        var now = ZonedDateTime.now(ZoneId.of("Z"));
         // @formatter:off
         return new KafkaProxyBuilder(primary)
                 .editOrNewStatus()
                     .withObservedGeneration(primary.getMetadata().getGeneration())
-                    .withConditions(effectiveReadyCondition(primary, exception))
+                    .withConditions(effectiveReadyCondition(primary, exception, now))
+                    .withClusters(clusterConditions(primary, context, now))
                 .endStatus()
             .build();
         // @formatter:on
     }
 
+    private static List<io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxystatus.Clusters> clusterConditions(KafkaProxy primary,
```

Comment:
We want the conditions to have a stable order (we don't want to end up generating a different list in successive reconciliations if nothing else has changed). I guess a Set would have a stable order, but it's not part of the `Set` contract. But using a List here and iterating in order (I guess I need a `.sequential()` in there to prevent the possibility of parallel stream processing) the clusters in the status will have the same order as the clusters in the spec ⇒ determinism.

However, all this will be changing with the factoring out of a `KafkaRoute` CR, so I don't want to expend efforts on tests for this now if I'm about to change it in the next PR.